### PR TITLE
Random points on lines update

### PR DESCRIPTION
**Pull request by havatv**
## Description

1. Update the progress reporting for the *Random Points on Lines* algorithm.
   * Follow the same logic as @nyalldawson suggested.
   * Use addition instead of multiplication.

   Seems to work correctly.

### CODE REVIEW

**Comment by nyalldawson** [comment]
@havatv can you rebase off master now and see the if the enum issue is fixed?

File: python/plugins/processing/gui/AlgorithmDialog.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ def runAlgorithm(self):
             self.iterateParam = None
 
             for param in self.algorithm().parameterDefinitions():
-                if isinstance(parameters.get(param.name(), None), QgsProcessingFeatureSourceDefinition) and parameters[param.name()].flags & QgsProcessingFeatureSourceDefinition.Flag.FlagCreateIndividualOutputPerInputFeature:
+                if isinstance(parameters.get(param.name(), None), QgsProcessingFeatureSourceDefinition) and parameters[param.name()].flags & QgsProcessingFeatureSourceDefinition.Flag.FlagCreateIndividualOutputPerInputFeature.value:
```
Comment (havatv) [inline python/plugins/processing/gui/AlgorithmDialog.py:211]:
```suggestion
                if isinstance(parameters.get(param.name(), None), QgsProcessingFeatureSourceDefinition) and parameters[param.name()].flags & QgsProcessingFeatureSourceDefinition.Flag.FlagCreateIndividualOutputPerInputFeature:
```

**Comment by havatv** [comment]
> @havatv can you rebase off master now and see the if the enum issue is fixed?

After `sudo apt-get install protobuf-compiler libprotoc-dev`, building and compiling went OK, and the ***Random points on lines*** algorithm works like a charm, even on selected features, and the progress bar behaves perfectly.
Thanks, @nyalldawson!

**Comment by nyalldawson** [comment]
Ouch - this PR has gotten in a mess. You'll probably need to open a new one....

**Comment by havatv** [comment]
So rebasing was probably not a good idea here... At least with my level of github expertise...
I will try to redo.

**Comment by havatv** [comment]
> Ouch - this PR has gotten in a mess. You'll probably need to open a new one....

New one opened: #35589.

**Pull request closed without merging**